src/main/java/uk/gov/hmcts/reform/divorce/orchestration/framework/workflow/task/TaskContextHelper.java

```
@@ -0,0 +1,40 @@
+package uk.gov.hmcts.reform.divorce.orchestration.framework.workflow.task;
+
+import lombok.AccessLevel;
+import lombok.NoArgsConstructor;
+import uk.gov.hmcts.reform.divorce.orchestration.domain.model.documentgeneration.GeneratedDocumentInfo;
+
+import java.util.HashSet;
+import java.util.List;
+
+import static uk.gov.hmcts.reform.divorce.orchestration.domain.model.OrchestrationConstants.DOCUMENT_COLLECTION;
+
+@NoArgsConstructor(access = AccessLevel.PRIVATE)
+public class TaskContextHelper {
+
+    public static void addToContextDocumentCollection(TaskContext context, GeneratedDocumentInfo documentInfoWithMetadata) {
+        context.computeTransientObjectIfAbsent(DOCUMENT_COLLECTION, new HashSet<>()).add(documentInfoWithMetadata);
+    }
+
+    public static void addAllToContextDocumentCollection(TaskContext context, List<GeneratedDocumentInfo> generatedDocumentInfoList)
+        throws InvalidDataForTaskException {
+        HashSet<GeneratedDocumentInfo> documentCollection = context.computeTransientObjectIfAbsent(DOCUMENT_COLLECTION, new HashSet<>());
+
+        generatedDocumentInfoList.forEach(documentInfo -> {
+            if (isNoneExistingDocument(documentCollection, documentInfo)) {
```

I think there's a typo. Should it be NonExisting?
Also, I don't think the isNoneExistingDocument method should throw the exception itself. You should throw the exception if this method returns false.
I think the method that should throw the exception is the first one (line 15).

---

Typo. Refactored and updated